tests/logic/test_pipelines_logic.py

```diff
@@ -67,113 +79,185 @@ def test_get_pipeline_info(mock_pipelines_api):
+    mock_input2.is_required = True
+    mock_input3 = mock()
+    mock_input3.name = OPTIONAL_INPUT_KEY
+    mock_input3.type = INTEGER_TYPE_KEY
+    mock_input3.is_required = False
+
+    mock_pipeline_info.inputs = [mock_input1, mock_input2, mock_input3]
+
+    when(pipelines_logic).get_pipeline_info(
+        TEST_PIPELINE_NAME, TEST_VERSION
+    ).thenReturn(mock_pipeline_info)
+
+    if error_messages:
+        with pytest.raises(SystemExit):
+            pipelines_logic.validate_pipeline_inputs(
+                TEST_PIPELINE_NAME, TEST_VERSION, input
+            )
+        for message in error_messages:
+            assert message in capture_logs.text
+    else:
+        # Should succeed with valid inputs
         pipelines_logic.validate_pipeline_inputs(
-            test_pipeline_name, None, {"input1": "value1"}
+            TEST_PIPELINE_NAME, TEST_VERSION, input
         )
 
-    assert "Missing or invalid inputs provided" in capture_logs.text
-    assert "Missing required input `input2`" in capture_logs.text
 
-
-def test_validate_pipeline_inputs_missing_file(capture_logs):
-    test_pipeline_name = "test_pipeline"
+validate_pipeline_inputs_file_testdata = [
+    # input dict, file_to_create (or None), list(error messages) if failure (None = validates)
+    ({STRING_INPUT_KEY: "foo", FILE_INPUT_KEY: "test_file"}, "test_file", None),
+    # failures:
+    (
+        {STRING_INPUT_KEY: "foo", FILE_INPUT_KEY: "test_file"},
+        None,
+        [
+            f"Error: Could not find provided file for input '{FILE_INPUT_KEY}': 'test_file'."
+        ],
+    ),
+    (
+        {STRING_INPUT_KEY: "foo", FILE_INPUT_KEY: None},
+        None,
+        [f"Error: Missing value for input '{FILE_INPUT_KEY}'."],
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    "input,file_to_create,error_messages", validate_pipeline_inputs_file_testdata
+)
+def test_validate_pipeline_inputs_file(
+    input: dict, file_to_create: str, error_messages: list[str], capture_logs
+):
+    # define mock input definitions
     mock_pipeline_info = mock()
-    mock_file_input = mock()
-    mock_file_input.name = "file_input"
-    mock_file_input.type = "FILE"
-    mock_pipeline_info.inputs = [mock_file_input]
+    mock_input1 = mock()
+    mock_input1.name = STRING_INPUT_KEY
+    mock_input1.type = STRING_TYPE_KEY
+    mock_input1.is_required = True
+    mock_input2 = mock()
+    mock_input2.name = FILE_INPUT_KEY
+    mock_input2.type = FILE_TYPE_KEY
+    mock_input2.is_required = True
 
-    when(pipelines_logic).get_pipeline_info(test_pipeline_name, 0).thenReturn(
-        mock_pipeline_info
-    )
+    mock_pipeline_info.inputs = [mock_input1, mock_input2]
 
-    with pytest.raises(SystemExit):
+    when(pipelines_logic).get_pipeline_info(
+        TEST_PIPELINE_NAME, TEST_VERSION
+    ).thenReturn(mock_pipeline_info)
+
+    if file_to_create:
+        # write the file locally
+        with open(file=file_to_create, mode="w") as blob_file:
```

Comment:
might be a little cleaner so use context manager and [TempDirectory](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory).  you can write the test in `with Tempdirectory` block and generate the file depending on `file_to_create`  and it'll clean itself up on exit